### PR TITLE
Updata

### PR DESCRIPTION
**Pull request by Alexander8933**

### CODE REVIEW

File: src/test/java/com/home/eshop/dao/BikesDaoTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        bikesDaoTest = new BikesDao();
+        bikesDaoTest.loadData();
+
+    }
+
+    @Test
+    void loadData() {
+        int size1 = bikesDaoTest.loadData();
+        int size2 = bikesDaoTest.bikes.size();
+        assertEquals(size2, size1);
```
Comment (sIvanovKonstantyn) [inline src/test/java/com/home/eshop/dao/BikesDaoTest.java:23]: You should check if the method result equals to you test data, which you prepare in your test

Comment (sIvanovKonstantyn) [reply src/test/java/com/home/eshop/dao/BikesDaoTest.java:23]: use **AssertJ** to test collections

File: src/test/java/com/home/eshop/dao/BikesDaoTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class BikesDaoTest {
+    private BikesDao bikesDaoTest;
+
+    @BeforeEach
+    void prepare() {
+        bikesDaoTest = new BikesDao();
```
Comment (sIvanovKonstantyn) [inline src/test/java/com/home/eshop/dao/BikesDaoTest.java:14]: Separate test and main databases

File: src/test/java/com/home/eshop/dao/BikesDaoTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        Bike bike1 = bikesDaoTest.bikes.get(id);
+        int size1 = ++id;
+        int size2 = bikesDaoTest.bikes.size();
+        assertEquals(size2, size1);
+        assertSame(bike2, bike1);
+    }
+
+    @Test
+    void findOne() {
+        Bike bike1 = bikesDaoTest.findOne(0);
```
Comment (sIvanovKonstantyn) [inline src/test/java/com/home/eshop/dao/BikesDaoTest.java:54]: You should check if the method result equals to you test data, which you prepare in your test

File: src/test/java/com/home/eshop/model/BikeTest.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class BikeTest {
+
+    private Bike bikeTest;
+
+    @BeforeEach
+    void prepare() {
+        bikeTest = new Bike("test", 1, 1);
```
Comment (sIvanovKonstantyn) [inline src/test/java/com/home/eshop/model/BikeTest.java:15]: move it to a fields definition

File: src/test/java/com/home/eshop/model/BikeTest.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    @AfterEach
+    void after() {
+        bikeTest = new Bike("test", 1, 1);
```
Comment (sIvanovKonstantyn) [inline src/test/java/com/home/eshop/model/BikeTest.java:20]: Useless. Remove

File: src/test/java/com/home/eshop/model/BikeTest.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    @Test
+    void getTextBikePresentation() {
+        String bike1 = "Title : " + "test" + " Price : " + "1" + " Number : " + "1";
+        String bike2 = bikeTest.getTextBikePresentation();
+        assertTrue(bike1.equals(bike2));
```
Comment (sIvanovKonstantyn) [inline src/test/java/com/home/eshop/model/BikeTest.java:27]: assertEquals

File: src/test/java/com/home/eshop/model/BikeTest.java [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    @Test
+    void getNumber() {
+        int number1 = 1;
+        int number2 = bikeTest.getNumber();
+        assertEquals(number1, number2);
+    }
+
+    @Test
+    void getPrice() {
+        int price1 = 1;
```
Comment (sIvanovKonstantyn) [inline src/test/java/com/home/eshop/model/BikeTest.java:77]: Move all getters and setters into one test case

File: src/test/java/com/home/eshop/dao/BikesDaoTest.java (hunk 1 above)
Comment (sIvanovKonstantyn) [reply src/test/java/com/home/eshop/dao/BikesDaoTest.java:23]: add fail tests case

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.home.eshop</groupId>
+    <artifactId>EbikeShop</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    <properties>
+        <maven.compiler.source>19</maven.compiler.source>
+        <maven.compiler.target>19</maven.compiler.target>
```
Comment (sIvanovKonstantyn) [inline pom.xml:13]: changes it to a last LTS version

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        <dependency>
+            <groupId>org.junit.platform</groupId>
+            <artifactId>junit-platform-suite</artifactId>
+            <version>${junit.platform.version}</version>
+            <scope>test</scope>
+        </dependency>
+
+
+    </dependencies>
+
```
Comment (sIvanovKonstantyn) [inline pom.xml:48]: add plugins and packaging section to build the project.

File: src/test/java/com/home/eshop/dao/BikesDaoTest.java (hunk 2 above)
Comment (Alexander8933) [reply src/test/java/com/home/eshop/dao/BikesDaoTest.java:14]: Implemented

File: src/test/java/com/home/eshop/dao/BikesDaoTest.java (hunk 3 above)
Comment (Alexander8933) [reply src/test/java/com/home/eshop/dao/BikesDaoTest.java:54]: implemented

File: src/test/java/com/home/eshop/model/BikeTest.java (hunk 5 above)
Comment (Alexander8933) [reply src/test/java/com/home/eshop/model/BikeTest.java:20]: implemented

File: src/test/java/com/home/eshop/model/BikeTest.java (hunk 6 above)
Comment (Alexander8933) [reply src/test/java/com/home/eshop/model/BikeTest.java:27]: implemented

File: src/test/java/com/home/eshop/model/BikeTest.java (hunk 7 above)
Comment (Alexander8933) [reply src/test/java/com/home/eshop/model/BikeTest.java:77]: implemented

File: pom.xml (hunk 8 above)
Comment (Alexander8933) [reply pom.xml]: implemented

File: pom.xml (hunk 9 above)
Comment (Alexander8933) [reply pom.xml:48]: implemented

File: src/test/java/com/home/eshop/dao/BikesDaoTest.java [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+        assertEquals(bikeOne.getTextBikePresentation(), bikeActual.getTextBikePresentation());
+    }
+
+    @Test
+    void saveAll() {
+        bikesDaoTest.saveAll();
+        bikes.clear();
+        bikesDaoTest.loadData(bikes);
+        bikeActual = bikes.get(0);
+        assertEquals(bikeOne.getTextBikePresentation(), bikeActual.getTextBikePresentation());
```
Comment (sIvanovKonstantyn) [inline src/test/java/com/home/eshop/dao/BikesDaoTest.java:48]: var bikesList = bikesDaoTest.findAll();
assertMatch(bikesList , containsAll(bikeOne , bikeTwo));

File: src/test/java/com/home/eshop/dao/BikesDaoTest.java [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+    BikesDao bikesDaoTest = new BikesDao(bikes, file);
+
+    Bike bikeOne = new Bike("testOne", 1, 1);
+    Bike bikeTwo = new Bike("testTwo", 2, 2);
+    Bike bikeActual;
+
+    @BeforeEach
+    void prepare() {
+        bikes.add(0, bikeOne);
+        bikes.add(1, bikeTwo);
```
Comment (sIvanovKonstantyn) [inline src/test/java/com/home/eshop/dao/BikesDaoTest.java:25]: bikesDaoTest.loadData(bikes);

File: src/test/java/com/home/eshop/dao/BikesDaoTest.java [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    @AfterEach
+    void complete() {
+        bikes.clear();
```
Comment (sIvanovKonstantyn) [inline src/test/java/com/home/eshop/dao/BikesDaoTest.java:30]: bikesDaoTest.loadData(bikes);

File: src/test/java/com/home/eshop/dao/BikesDaoTest.java (hunk 10 above)
Comment (Alexander8933) [reply src/test/java/com/home/eshop/dao/BikesDaoTest.java]: implemented

File: src/test/java/com/home/eshop/dao/BikesDaoTest.java (hunk 11 above)
Comment (Alexander8933) [reply src/test/java/com/home/eshop/dao/BikesDaoTest.java]: implemented

File: src/test/java/com/home/eshop/dao/BikesDaoTest.java (hunk 12 above)
Comment (Alexander8933) [reply src/test/java/com/home/eshop/dao/BikesDaoTest.java]: implemented

File: src/test/java/com/home/eshop/dao/BikesDaoTest.java (hunk 1 above)
Comment (Alexander8933) [reply src/test/java/com/home/eshop/dao/BikesDaoTest.java]: Implemented+-

**Pull request merged**